aspnetcore/fundamentals/minimal-apis/bindStreamPipeReader/7.0-samples/PipeStreamToBackgroundQueue/Program.cs

```diff
@@ -0,0 +1,31 @@
+using System.Threading.Channels;
+using BackgroundQueueService;
+
+var builder = WebApplication.CreateBuilder(args);
+// Create a channel to send data to the background queue.
+builder.Services.AddSingleton<Channel<ReadOnlyMemory<byte>>>((_) => Channel.CreateBounded<ReadOnlyMemory<byte>>(6400));
```

Comment:
OK lets add some comments and constants:

```suggestion
// The max memory we're willing to use for the upload endpoint on this instance
var maxMemory = 500 * 1024 * 1024;

// The max size of a single message, we're staying below the default LOH size of 85K
var maxMessageSize = 80 * 1024 * 1024;

// The max size of the queue based on those restrictions
var maxQueueSize = maxMemory / maxMessageSize;

// Create a channel to send data to the background queue.
builder.Services.AddSingleton<Channel<ReadOnlyMemory<byte>>>((_) => Channel.CreateBounded<ReadOnlyMemory<byte>>(maxQueueSize));
```

Reply:
Yh this is better.. would help the people copying the sample code to understand the given constraint

Reply:
but this leaves it at 6.25 bounded space which ought to be an int

Reply:
oops, fixed.

Reply:
okay cool

Reply:
saw the automatic type conversion 😅